docs/madr/decisions/040-meshservice-hostname-vips.md

```diff
@@ -0,0 +1,128 @@
+# Hostnames and VIP management in MeshService
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/6331
+
+## Context and Problem Statement
+
+With introduction of MeshService, we need to have a mechanism to assign VIPs and hostnames to MeshServices.
+Currently, VIPs are managed in a ConfigMap per Mesh. Hostnames are managed using VirtualOutbound.
+
+## Considered Options
+
+* VIPs management
+* Hostname Generator
+* Static Hostname template in CP config
+
+## Decision Outcome
+
+Chosen option: "Hostname Generator" and "VIPs management" since it's the most flexible option.
+
+## Pros and Cons of the Options
+
+### VIPs management
+
+We have to assign a VIP for each MeshService to be able to address it using transparent proxy.
+
+MeshService object contains VIPs in status.
+```yaml
+status: # managed by CP. Not shared cross zone, but synced to global
+  vips:
+  - ip: <kube_cluster_ip> # or kuma VIP
```

Comment:
What about headless service? These are pod Ips?

Reply:
Yes. I'll cover that in https://github.com/kumahq/kuma/pull/9722

Reply:
I think maybe it's worth covering here how we update status in the case of a headless MeshService

Reply:
added that it can be pod ip